modules/optimeraBidAdapter.js

```diff
@@ -0,0 +1,81 @@
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'optimera';
+const SCORES_BASE_URL = 'https://s3.amazonaws.com/elasticbeanstalk-us-east-1-397719490216/json/client/';
+
+export const spec = {
+  code: BIDDER_CODE,
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {bidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bidRequest) {
+    if (typeof bidRequest.params.custom != 'undefined' && typeof bidRequest.params.custom.clientID != 'undefined') {
+      return true;
+    } else {
+      return false;
+    }
+  },
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests) {
+    var optimeraHost = window.location.host;
+    var optimeraPathName = window.location.pathname;
+    var timestamp = Math.round(new Date().getTime() / 1000);
+    if (typeof validBidRequests[0].params.custom.clientID != 'undefined') {
+      var clientID = validBidRequests[0].params.custom.clientID;
+      var scoresURL = SCORES_BASE_URL + clientID + '/' + optimeraHost + optimeraPathName + '.js';
+      return {
+        method: 'GET',
+        url: scoresURL,
+        payload: validBidRequests,
+        data: {'t': timestamp}
+      };
+    }
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * Some required bid params are not needed for this so default
+   * values are used.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    var scores = serverResponse.body.replace('window.oVa = ', '');
+    scores = scores.replace(';', '');
+    scores = JSON.parse(scores);
+    var validBids = bidRequest.payload;
+    var bidResponses = [];
+    var dealId = '';
+    for (var i = 0; i < validBids.length; i++) {
+      if (typeof validBids[i].params.custom.clientID != 'undefined') {
+        if (validBids[i].adUnitCode in scores) {
+          dealId = scores[validBids[i].adUnitCode];
+        }
+        var bidResponse = {
+          requestId: validBids[i].bidId,
+          ad: '<div></div>',
```

Comment:
this might have been mentioned before, but how do you render the ad if there is no ad content?

Reply:
hey @mkendall07, we are just creating this adpator to add visibility scores for the ad divs that will be added to DFP calls.

Reply:
Ok but what does a publisher do with that information? And why is a request to your endpoint required to calculate viewability?